File: src/API/API.Application/DTOs/UserParams.cs
using API.Domain.Constants;
using API.Domain.Enums;

namespace API.Application.DTOs
{
    public class UserParams : PaginationParams
    {
        private int minAge = SystemPolicy.UsersMinimumAge;
        private int? maxAge = null;
        public SexOptions Sex { get; set; } = SexOptions.None; // to get the default value we need to make a query which is the interest field. 
        public OrderByOptions OrderBy { get; set; } = OrderByOptions.LastActive;
        public int MinAge
        {
            get
            {
                return minAge;
            }
            set
            {
                if (value >= SystemPolicy.UsersMinimumAge)
                {
                    minAge = value;
                }
            }
        }
        public int? MaxAge
        {
            get
            {
                return maxAge;
            }
            set
            {
                if (value >= minAge)
                {
                    maxAge = value;
                }
            }
        }
    }

    
}

File: src/Frontend [Deprecated]/src/app/_models/userParams.ts
import { User } from "./User";

export class UserParams{
    minAge :  number = 18;
    maxAge : null | number = null;
    pageNumber : number = 1;
    itemsPerPage: number = 16;
    orderBy: string = 'lastActive'; // options lastActive, age, creationTime
    sex: string;
    constructor( user: User){
        this.sex = user.interest;
    }
}